client/my-sites/upgrades/cart/cart-body.jsx

```diff
@@ -1,50 +1,56 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-var CartItems = require( './cart-items' ),
-	CartCoupon = require( './cart-coupon' ),
-	CartTotal = require( './cart-total' );
-
-var CartBody = React.createClass( {
-	propTypes: {
-		collapse: React.PropTypes.bool
-	},
-	getDefaultProps: function() {
-		return {
-			collapse: false,
-			showCoupon: false
-		};
-	},
-
-	render: function() {
-		if ( ! this.props.cart.hasLoadedFromServer ) {
-			return <div className="cart-body">{ this.translate( 'Loading…', { context: 'Upgrades: Loading cart' } ) }</div>;
+import CartItems from './cart-items';
+import CartCoupon from './cart-coupon';
+import CartTotal from './cart-total';
+
+class CartBody extends Component {
+	static propTypes = {
+		collapse: PropTypes.bool
+	};
+
+	static defaultProps = {
+		collapse: false,
+		showCoupon: false
+	};
+
+	optionalCoupon() {
+		if ( ! this.props.showCoupon ) {
+			return;
+		}
+
+		return <CartCoupon cart={ this.props.cart } />;
+	}
+
+	render() {
+		const { translate, cart } = this.props;
+
+		if ( ! cart.hasLoadedFromServer ) {
+			return <div className="cart-body">
+				{ translate( 'Loading…', { context: 'Upgrades: Loading cart' } ) }
```

Comment:
Good catch!  We only want a context if the word could be translated in different ways.  You can see this is the third instance with the same translation:
https://translate.wordpress.com/projects/wpcom/fr/default/?filters%5Bterm%5D=Loading%E2%80%A6&filters%5Buser_login%5D=&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Bview%5D=&filter=Filter&sort%5Bby%5D=priority&sort%5Bhow%5D=desc

This would make a great comment. If you're unsure, it's better to err on the side of a comment, and then translators will ping i18n and we can switch it to a comment for the cases that need it.